Automattic/WP-Job-Manager
# Adds core email notification handling

This is a large chunk of functionality that I'm adding. This is the largest/first PR that just adds an MVP and starts resolving #672 (as well as others later).

Additional PRs will be added to:
- Show job details in notification to admin when new jobs are created. 
- Finish the actual notification for #672.
- Add an administration interface for enabling/disabling each notification in settings.
- Add additional notifications for editing job listings and notice of expiring job listings (#446).
- I'm also going to create a PR for the Resume's plugin to test compatibility with a WPJM extension plugin.

This PR:
- Adds the core functionality for sending notifications in WPJM.
- Includes a very minimal version of a notification that will be sent to admins when a new job listing is created.
- Adds unit tests functionality included so-far.

## Testing Instructions
- Verify unit tests pass.
- On submitting a new job listing from the `[submit_job_form]` shortcode, verify a very basic email is sent to the WordPress admin when submitted.
